src/transformation/visitors/access.ts

```diff
@@ -31,19 +31,13 @@ export const transformElementAccessExpression: FunctionVisitor<ts.ElementAccessE
         return constEnumValue;
     }
 
-    const table = context.transformExpression(expression.expression);
-
     const argumentType = context.checker.getTypeAtLocation(expression.argumentExpression);
     const type = context.checker.getTypeAtLocation(expression.expression);
     if (isNumberType(context, argumentType) && isStringType(context, type)) {
-        const index = context.transformExpression(expression.argumentExpression);
-        return lua.createCallExpression(
-            lua.createTableIndexExpression(lua.createIdentifier("string"), lua.createStringLiteral("sub")),
-            [table, createExpressionPlusOne(index), createExpressionPlusOne(index)],
-            expression
-        );
+        return transformStringIndex(context, expression);
     }
 
+    const table = context.transformExpression(expression.expression);
     return lua.createTableIndexExpression(table, transformElementAccessArgument(context, expression), expression);
```

Comment:
That `table` variable name might be a bit confusing, since you don't necessarily access a property on a table. And in general the variable in this case doesn't seem to be very helpful here
```suggestion
    return lua.createTableIndexExpression(
        context.transformExpression(expression.expression),
        transformElementAccessArgument(context, expression),
        expression
     );
```